### ISSUE TITLE
Responsive CSS

### ISSUE THREAD

**Issue by iangreenleaf** [grinnellplans/GrinnellPlans#132]
Short-term goals:

 * Have a decent default small-screen viewing experience so that Plans is functional on mobile
 * Serve this to all small-screen clients (under a certain size limit, perhaps?)

Long-term goals (maybe? not set in stone):

 * Support mobile responsiveness for different stylesheets
 * Support responsive images and media queries and whatever other mobile enhancements will make life better
 * Allow Plans to easily adapt to future changes in the display landscape

**Comment by hannaliebl** [comment]
I'm happy to look into this. I can't give a good estimate now, but in a few hours I'll poke around and give my impression. My initial feeling is that this is doable before release.

**Comment by jamesprior** [comment]
Yeah, I was hoping that we could add basic bootstrap styles to all the templates and only serve the core bootstrap CSS to clients we can identify as being mobile. My thought was to keep the same view templates for mobile and desktop without breaking the existing desktop CSS. I'm not sure if that plan would work though, just an idea. 

--James

**Comment by hannaliebl** [comment]
Bootstrap is mobile first by default, though, so it should work regardless. Any modern CSS grid system will use media queries and sane defaults, so I think, personally, conditionally-served CSS is unnecessary (though maybe within the context of this project it is?) I forgot, I made [this](http://www.hannaliebl.com/lab/mobile-plans/) a long time ago using the existing markup plans had at the time, and so assuming the media-viewport tag is there (and I used some jquery for the menus I think), responsive styles are easy enough to add, even with just one media query. I'm sorry, also, I didn't really do a good job looking into this last night - I will give myself the upcoming weekend to investigate, if that's ok.

**Comment by iangreenleaf** [comment]
After discussing this with a bunch of people, I've come up with a rough plan of action. I've opened a new milestone to track these steps and will be breaking out the specific tasks into tickets there. I think I'll leave this open as kind of a place for general discussion, at least for a while.

Here's the game plan, in order:

1. Drop old interfaces, port old stylesheets to new interface (#92)
2. Add meta viewport tag (#89)
3. Make changes to HTML layout to support responsive design (is anything needed here?)
4. Provide simple responsive CSS defaults that most stylesheets can use (#139)
5. Pull updates from #4 into all the core Plans styles (#141)
6. Do some outreach for other custom stylesheet authors (#140)
6. Make a new default stylesheet for new plans users (#142)
